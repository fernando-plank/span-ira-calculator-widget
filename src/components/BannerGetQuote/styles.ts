import styled from 'styled-components'

export const Wrapper = styled.div`
  width: 100%;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    background: url(${({ theme }) => theme.images.quote.mobile}) no-repeat;
    color: ${({ theme }) => theme.colors.black};
    background-size: 100%;
    height: auto;
    padding-top: 40px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.tablet}) {
    background: url(${({ theme }) => theme.images.quote.desktop}) no-repeat
      center;
    color: ${({ theme }) => theme.colors.white};
    background-size: 100%;
    height: 730px;
    padding-top: 0px;
  }
`

export const BannerImage = styled.img`
  width: 100%;
`

export const BannerContent = styled.div`
    text-align: left;
    display: flex;

    @media(min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
        padding 41px 22px;
        flex-direction: column;
        margin-top: 278px;
    }

    @media(min-width: ${({ theme }) => theme.breakpoints.tablet}) {
        padding 60px 55px;
        flex-direction: row;
        align-items: flex-end;
        padding-top: 245px;
        height: 640px;
        margin-top: 0px;
    }
`

export const BannerTextWrapper = styled.div`
  @media (min-width: ${({ theme }) => theme.breakpoints.tablet}) {
    width: 70%;
    align-self: center;
  }
`

export const BannerTitle = styled.h1`
  font-weight: 600;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    font-size: 70px;
    line-height: 70px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.tablet}) {
    font-size: 136px;
    line-height: 136px;
  }
`

export const BannerDescription = styled.p`
  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    font-weight: 500;
    font-size: 16px;
    line-height: 24px;
    margin-bottom: 51px;
    margin-top: 40px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.tablet}) {
    width: 50%;
    margin-top: 63px;
  }
`

export const GetQuoteButton = styled.button`
  cursor: pointer;
  border-radius: 20px;
  margin-right: 14px;
  height: 42px;
  width: 50%;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    font-weight: 700;
    font-size: 16px;
    line-height: 24px;
    background-color: ${({ theme }) => theme.colors.black};
    border: 2px solid ${({ theme }) => theme.colors.black};
    color: ${({ theme }) => theme.colors.white};
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.tablet}) {
    font-weight: 700;
    font-size: 16px;
    line-height: 24px;
    background-color: ${({ theme }) => theme.colors.white};
    border: 2px solid ${({ theme }) => theme.colors.white};
    color: ${({ theme }) => theme.colors.black};
  }
`

export const ExplorePanelButton = styled.button`
  cursor: pointer;
  color: ${({ theme }) => theme.colors.black};
  margin-left: 14px;
  height: 42px;
  width: 50%;
  background: none;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    font-weight: 700;
    font-size: 16px;
    line-height: 24px;
    border: 2px solid ${({ theme }) => theme.colors.black};
    border-radius: 20px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.tablet}) {
    font-weight: 700;
    font-size: 16px;
    line-height: 24px;
    border: 2px solid ${({ theme }) => theme.colors.white};
    color: ${({ theme }) => theme.colors.white};
  }
`

export const ButtonWrapper = styled.div`
  display: flex;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    font-weight: 700;
    font-size: 16px;
    line-height: 24px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.tablet}) {
    width: 30%;
  }
`
