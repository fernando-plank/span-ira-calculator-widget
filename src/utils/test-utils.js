var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
/* eslint-disable import/export */
import React from 'react';
import theme from '@styles/theme';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
jest.mock('../components/Icon', () => {
    return jest.fn(() => React.createElement("div", null, "mocked icon"));
});
const customRender = (ui, _a = {}) => {
    var renderOptions = __rest(_a, []);
    return render(React.createElement(ThemeProvider, { theme: theme }, ui), renderOptions);
};
export * from '@testing-library/react';
export { customRender as render };
