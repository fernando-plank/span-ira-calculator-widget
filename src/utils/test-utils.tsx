/* eslint-disable import/export */
import { ReactElement } from 'react'
import { RenderOptions, render } from '@testing-library/react'

import theme from '@styles/theme'
import { ThemeProvider } from 'styled-components'

jest.mock('../components/Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

const customRender = (ui: ReactElement, { ...options }: RenderOptions = {}) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options)

export * from '@testing-library/react'
export { customRender as render }
