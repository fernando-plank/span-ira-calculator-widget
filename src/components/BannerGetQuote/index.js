import React from 'react';
import * as S from './styles';
const BannerGetQuote = () => (React.createElement(S.Wrapper, null,
    React.createElement(S.BannerContent, null,
        React.createElement(S.BannerTextWrapper, null,
            React.createElement(S.BannerTitle, null, "Ready"),
            React.createElement(S.BannerTitle, null, "for SPAN?"),
            React.createElement(S.BannerDescription, null, "Complete our easy form and one of our Home Advisors will be in touch to discuss your project scope, and the savings available.")),
        React.createElement(S.ButtonWrapper, null,
            React.createElement(S.GetQuoteButton, null, "Get quote"),
            React.createElement(S.ExplorePanelButton, { onClick: () => window.open('https://www.span.io/panel', '_self') }, "Explore panel")))));
export default BannerGetQuote;
