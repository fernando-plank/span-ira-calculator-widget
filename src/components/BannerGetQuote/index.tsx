import React from 'react'
import * as S from './styles'

const BannerGetQuote = () => (
  <S.Wrapper>
    <S.BannerContent>
      <S.BannerTextWrapper>
        <S.BannerTitle>Ready</S.BannerTitle>
        <S.BannerTitle>for SPAN?</S.BannerTitle>
        <S.BannerDescription>
          Complete our easy form and one of our Home Advisors will be in touch
          to discuss your project scope, and the savings available.
        </S.BannerDescription>
      </S.BannerTextWrapper>
      <S.ButtonWrapper>
        <S.GetQuoteButton>Get quote</S.GetQuoteButton>
        <S.ExplorePanelButton
          onClick={() => window.open('https://www.span.io/panel', '_self')}
        >
          Explore panel
        </S.ExplorePanelButton>
      </S.ButtonWrapper>
    </S.BannerContent>
  </S.Wrapper>
)

export default BannerGetQuote
