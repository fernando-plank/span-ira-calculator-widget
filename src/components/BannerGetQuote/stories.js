import React from 'react';
import BannerGetQuote from '.';
export default {
    title: 'BannerGetQuote',
    component: BannerGetQuote
};
export const Default = () => React.createElement(BannerGetQuote, null);
