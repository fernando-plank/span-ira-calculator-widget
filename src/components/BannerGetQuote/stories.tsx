import React from 'react'
import { Story, Meta } from '@storybook/react'

import BannerGetQuote from '.'

export default {
  title: 'BannerGetQuote',
  component: BannerGetQuote
} as Meta

export const Default: Story = () => <BannerGetQuote />
