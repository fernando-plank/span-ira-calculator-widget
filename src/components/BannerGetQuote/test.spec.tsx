import { screen } from '@testing-library/react'
import { render } from '@utils/test-utils'

import BannerGetQuote from '.'
import React from 'react'

describe('<BannerGetQuote />', () => {
  // TODO Tests for buttons actions
  it('should render explore panel button', () => {
    const { container } = render(<BannerGetQuote />)

    expect(screen.getByText('Explore panel')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render get quote button', () => {
    const { container } = render(<BannerGetQuote />)

    expect(screen.getByText('Get quote')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
