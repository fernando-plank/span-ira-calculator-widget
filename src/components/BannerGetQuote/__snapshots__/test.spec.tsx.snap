// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BannerGetQuote /> should render explore panel button 1`] = `
<div
  class="sc-bgqQcB kOiKDX"
>
  <div
    class="sc-ewnqHT ecJhSW"
  >
    <div
      class="sc-fFGjHI cQAWRu"
    >
      <h1
        class="sc-dicizt bxdNUB"
      >
        Ready
      </h1>
      <h1
        class="sc-dicizt bxdNUB"
      >
        for SPAN?
      </h1>
      <p
        class="sc-iVCKna gTFYEs"
      >
        Complete our easy form and one of our Home Advisors will be in touch to discuss your project scope, and the savings available.
      </p>
    </div>
    <div
      class="sc-ispOId iBCoDe"
    >
      <button
        class="sc-gxYJeL dPPtxZ"
      >
        Get quote
      </button>
      <button
        class="sc-kPTPQs dbMuJB"
      >
        Explore panel
      </button>
    </div>
  </div>
</div>
`;

exports[`<BannerGetQuote /> should render get quote button 1`] = `
<div
  class="sc-bgqQcB kOiKDX"
>
  <div
    class="sc-ewnqHT ecJhSW"
  >
    <div
      class="sc-fFGjHI cQAWRu"
    >
      <h1
        class="sc-dicizt bxdNUB"
      >
        Ready
      </h1>
      <h1
        class="sc-dicizt bxdNUB"
      >
        for SPAN?
      </h1>
      <p
        class="sc-iVCKna gTFYEs"
      >
        Complete our easy form and one of our Home Advisors will be in touch to discuss your project scope, and the savings available.
      </p>
    </div>
    <div
      class="sc-ispOId iBCoDe"
    >
      <button
        class="sc-gxYJeL dPPtxZ"
      >
        Get quote
      </button>
      <button
        class="sc-kPTPQs dbMuJB"
      >
        Explore panel
      </button>
    </div>
  </div>
</div>
`;
